### DISCUSSION TITLE
Doc comment on text2vec-transformers (/modules/retriever-vectorizer-modules/text2vec-transformers)

### DISCUSSION THREAD

**Discussion by giscus** [weaviate/weaviate-io#751] [Docs/blog comments]
# Doc comment on text2vec-transformers (/modules/retriever-vectorizer-modules/text2vec-transformers)

Introduction

https://weaviate.io/developers/weaviate/modules/retriever-vectorizer-modules/text2vec-transformers

**Comment by scepeda78** [comment]
if a custom vectorized is used, then for the query WCS would need that we have the local server with the vectorized running, to vectorize the query?

**Comment by hbsardhara09** [comment]
How to create a vector using "text2vec-transformers" module? I want to use "nearVector" method but there is no way to create vector using "text2vec-transformers" module. I am using this module as my default vectorizer.

**Comment by peterhessey** [comment]
When trying to run the docker compose file listed under "Option 1: Use a pre-built transformer model container", I get the following error if try to set `ENABLE_CUDA: 1`:

```shell
embeddings-t2v-transformers-1  | RuntimeError: Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx
embeddings-t2v-transformers-1  | 
embeddings-t2v-transformers-1  | ERROR:    Application startup failed. Exiting.
```

Any ideas on what is causing this? NVIDIA and CUDA are set up on properly on my machine (confired with `nvidia-smi`.

**Reply by peterhessey** [reply]
confirmed with `nvidia-smi`)*

**Reply by zachtrong** [reply]
Try this, expose gpus from host to docker container, it worked for me with docker on Windows WSL.
https://stackoverflow.com/questions/70761192/docker-compose-equivalent-of-docker-run-gpu-all-option

**Comment by daviibf** [comment]
Within a same schema, a have two different topic-related classes that are referenced one by the other. I was willing to vectorize one of those classes using a pre-built text2vec transformer image, at the same time I vectorize the other using a custom text2vec transformer image. Is it possible to do it ?

**Comment by mikeloh77** [comment]
I am following the steps for Option 2. Unfortunately when testing the vectorizer 
```
curl localhost:9090/vectors -H 'Content-Type: application/json' -d '{"text": "foo bar"}'
```
I get a 404 error
```
{"code":404,"message":"path /vectors was not found"}%       
```
I think there must be a bug in the custom image or the documentation is out of date. As I have no way of inspecting the available routes I can not confirm this or figure out a solution. Please help.

**Reply by dudanogueira** [reply]
Hi! You need to add
```
    ports:
    - 9090:8080
```

to the `t2v-transformers` docker service, Like so:
```
  t2v-transformers:
    image: semitechnologies/transformers-inference:sentence-transformers-multi-qa-MiniLM-L6-cos-v1
    environment:
      ENABLE_CUDA: 0 # set to 1 to enable
    ports:
    - 9090:8080
```

**Comment by khaddeep143** [comment]
Works fine but if run on many data, I am getting below error {'error': [{'message': 'update vector: send POST request: Post "http://t2v-transformers:8080/vectors": dial tcp 172.19.0.2:8080: connect: connection refused'}]}
